### PR TITLE
Fix LSP server crash when `".` is inserted into the editor

### LINKED ISSUE #958: LSP server crashes on `"` followed by a `.` for completion
**Describe the bug**
When  `".` is typed in the editor, the LSP server crashes. 

**To Reproduce**
Open an editor with the Nickel LSP server attached and running and type `".`.

**Expected behavior**
Nothing, the LSP should not crash.

**Additional context**
The crash error message: 
```
thread 'main' panicked at 'begin <= end (1 <= 0) when slicing `"`', lsp/nls/src/requests/completion.rs:267:27
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[Info  - 17:39:21] Connection to server got closed. Server will restart.
[Error - 17:39:21] Request textDocument/completion failed.
```
See [here.](https://github.com/tweag/nickel/blob/master/lsp/nls/src/requests/completion.rs#L169)

### PR DESCRIPTION
**Pull request by ebresafegaga**
Closes #958

### CODE REVIEW

**Review by matthew-healy — APPROVED** [review]
:+1:

**Review by matthew-healy — APPROVED** [review]
Okay, I had got a little confused by the `".` in the title. The problem was really if the name was just a single `"` as then the first and last char of the string would be the same, which is why using a single iterator fixes it. Approving with confidence this time. :grin:

File: lsp/nls/src/requests/completion.rs [hunk 1]
```diff
@@ -166,7 +166,8 @@ fn get_identifier_path(text: &str) -> Option<Vec<String>> {
     let path = result.iter().rev().next().cloned()?;
     // Remove quotes from a record name
     fn remove_quotes(name: &str) -> String {
-        if let (Some('"'), Some('"')) = (name.chars().next(), name.chars().last()) {
+        let mut chars = name.chars();
+        if let (Some('"'), Some('"')) = (chars.next(), chars.last()) {
```
Comment (sir4ur0n) [inline lsp/nls/src/requests/completion.rs:170]: Can we add a test as evidence of this fix? :pray: I.e. a test that fails before the fix, and succeeds after the fix

**Review by sir4ur0n — CHANGES_REQUESTED** [review]

File: lsp/nls/src/requests/completion.rs (hunk 1 above)
Comment (ebresafegaga) [reply lsp/nls/src/requests/completion.rs:170]: Done, thanks!

**Review by sir4ur0n — APPROVED** [review]

**Review by dpl0a — APPROVED** [review]

**Pull request merged**